monai/apps/detection/utils/anchor_utils.py

```diff
@@ -0,0 +1,409 @@
+            ``scale[i]*base_anchor_shapes`` represents the anchor shapes for feature map ``i``.
+        base_anchor_shapes: a sequence which represents several anchor shapes for one feature map.
+            For N-D images, it is a Sequence of N value Sequence.
+        indexing: choose from {'xy', 'ij'}, optional
+            Cartesian ('xy') or matrix ('ij', default) indexing of output.
+            Cartesian ('xy') indexing swaps axis 0 and 1, which is the setting inside torchvision.
+            matrix ('ij', default) indexing keeps the original axis not changed.
+            See also indexing in https://pytorch.org/docs/stable/generated/torch.meshgrid.html
+
+    Example:
+        .. code-block:: python
+
+            # 2D example inputs for a 2-level feature maps
+            feature_map_scales = (1, 2)
+            base_anchor_shapes = ((10, 10), (6, 12), (12, 6))
+            AnchorGeneratorWithAnchorShape(feature_map_scales, base_anchor_shapes)
+
+            # 3D example inputs for a 2-level feature maps
+            feature_map_scales = (1, 2)
+            base_anchor_shapes = ((10, 10, 10), (12, 12, 8), (10, 10, 6), (16, 16, 10))
+            AnchorGeneratorWithAnchorShape(feature_map_scales, base_anchor_shapes)
+    """
+
+    __annotations__ = {"cell_anchors": List[torch.Tensor]}
+
+    def __init__(
+        self,
+        feature_map_scales: Union[Sequence[int], Sequence[float]] = (1, 2, 4, 8),
+        base_anchor_shapes: Union[Sequence[Sequence[int]], Sequence[Sequence[float]]] = (
+            (32, 32, 32),
+            (48, 20, 20),
+            (20, 48, 20),
+            (20, 20, 48),
+        ),
+        indexing: str = "ij",
+    ) -> None:
+        nn.Module.__init__(self)
+
+        spatial_dims = len(base_anchor_shapes[0])
+        spatial_dims = look_up_option(spatial_dims, [2, 3])
+        self.spatial_dims = spatial_dims
+
+        self.indexing = look_up_option(indexing, ["ij", "xy"])
+
+        base_anchor_shapes_t = torch.Tensor(base_anchor_shapes)
+        self.cell_anchors = [self.generate_anchors_using_shape(s * base_anchor_shapes_t) for s in feature_map_scales]
+
+    @staticmethod
+    def generate_anchors_using_shape(
+        anchor_shapes: torch.Tensor, dtype: torch.dtype = torch.float32, device: Union[torch.device, None] = None
+    ) -> torch.Tensor:
+        """
+        Compute cell anchor shapes at multiple sizes and aspect ratios for the current feature map.
+
+        Args:
+            anchor_shapes: [w, h] or [w, h, d], sized (N, spatial_dims),
+                represents N anchor shapes for the current feature map.
+            dtype: target data type of the output Tensor.
+            device: target device to put the output Tensor data.
+
+        Returns:
+            For 2D images, returns [-w/2, -h/2, w/2, h/2];
+            For 3D images, returns [-w/2, -h/2, -d/2, w/2, h/2, d/2]
+        """
+        if device is None:
+            device = torch.device("cpu")
```

Comment:
Maybe we don't need to change the device if `device=None`.
Default to `cpu` is easy to cause errors.
You can refer to:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/inferers/utils.py#L98

Thanks.